File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    answer = []
    stack = []
    
    i, j = 1, 0
    
    while i <= len(order):
        while stack and stack[-1] == order[j]:
            answer.append(stack.pop())
            j += 1
            
        if i != order[j]:
            stack.append(i)
        else:
            answer.append(i)
            j += 1
            
        i += 1
        
    while stack and stack[-1] == order[j]:
        answer.append(stack.pop())
        j += 1
    
    return len(answer)

File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;

class Solution {
    
    public int solution(int[] order) {
        List<Integer> answer = new ArrayList<>();
        Stack<Integer> stack = new Stack<>();
        
        int i = 1;
        int j = 0;
        
        while (i <= order.length) {
            while (!stack.isEmpty() && stack.peek() == order[j]) {
                answer.add(stack.pop());
                j++;
            }
            
            if (i != order[j]) {
                stack.push(i);
            } else {
                answer.add(i);
                j++;
            }
            
            i++;
        }
        
        while (!stack.isEmpty() && stack.peek() == order[j]) {
            answer.add(stack.pop());
            j++;
        }
        
        return answer.size();
    }
}